packages/api-plugin-fulfillment/migrations/4.js

```diff
@@ -0,0 +1,64 @@
+const affectedGlobalGroups = [
+  "owner",
+  "shop manager"
+];
+
+const newGlobalPermissions = [
+  "reaction:legacy:fulfillmentRestrictions/create",
+  "reaction:legacy:fulfillmentRestrictions/delete",
+  "reaction:legacy:fulfillmentRestrictions/read",
+  "reaction:legacy:fulfillmentRestrictions/update",
+  "reaction:legacy:fulfillmentTypes/create",
+  "reaction:legacy:fulfillmentTypes/delete",
+  "reaction:legacy:fulfillmentTypes/read",
+  "reaction:legacy:fulfillmentTypes/update",
+  "reaction:legacy:fulfillmentMethods/create",
+  "reaction:legacy:fulfillmentMethods/delete",
+  "reaction:legacy:fulfillmentMethods/read",
+  "reaction:legacy:fulfillmentMethods/update"
+];
+
+/**
+ * @summary migrates the database down one version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function down({ db, progress }) {
+  progress(0);
+
+  await db.collection("Groups").updateMany({
+    slug: { $in: affectedGlobalGroups }
+  }, {
+    $pullAll: { permissions: newGlobalPermissions }
+  });
+
+  progress(100);
+}
+
+/**
+ * @summary Performs migration up from previous data version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function up({ db, progress }) {
+  progress(0);
+
+  await db.collection("Groups").updateMany({
```

Comment:
This just adds them to shop manager and owner which is fine if they used the Out of the Box permissions and made no changes, but what it actually should do is add it to any group that has the old equivalent permissions?

Reply:
@zenweasel 

When we say "_any group that has the old equivalent permissions_", does it mean any group which would have the same entries in the array as of shop-manager and owner? 

If the user has made a minor add/remove of any entry, then how do we decide whether that groups has to be considered or not?

Reply:
@zenweasel 
As discussed, I have made the code change to update 'any' group that has the permissions. 
As per the change, we are checking each group for any of the  old permissions (related to shipping) and inserting the corresponding individual FFT permissions (using a mapper) in all of the impacted groups.